traffic_ops/traffic_ops_golang/server/servers.go

```diff
@@ -359,267 +737,482 @@ WHERE s.id IN (` + edgeIDs + `)))
 	}
 
-	dsIDs := []int64{}
-	if err := s.APIInfo().Tx.QueryRowx("SELECT ARRAY(SELECT deliveryservice FROM deliveryservice_server WHERE server = $1)", s.ID).Scan(pq.Array(&dsIDs)); err != nil && err != sql.ErrNoRows {
+	var dsIDs []int64
+	if err := tx.QueryRowx("SELECT ARRAY(SELECT deliveryservice FROM deliveryservice_server WHERE server = $1)", server.ID).Scan(pq.Array(&dsIDs)); err != nil && err != sql.ErrNoRows {
 		return nil, fmt.Errorf("getting server assigned delivery services: %v", err), http.StatusInternalServerError
 	}
+	// If type is changing ensure it isn't assigned to any DSes.
+	if typeID != *server.TypeID {
+		if len(dsIDs) != 0 {
+			return errors.New("server type can not be updated when it is currently assigned to Delivery Services"), nil, http.StatusConflict
+		}
+	}
 	// Check to see if the user is trying to change the CDN of a server, which is already linked with a DS
-	if cdnId != *s.CDNID && len(dsIDs) != 0 {
+	if cdnID != *server.CDNID && len(dsIDs) != 0 {
 		return errors.New("server cdn can not be updated when it is currently assigned to delivery services"), nil, http.StatusConflict
 	}
-	// If type is changing ensure it isn't assigned to any DSes.
-	if typeID != *s.TypeID {
-		if len(dsIDs) != 0 {
-			return errors.New("server type can not be updated when it is currently assigned to delivery services"), nil, http.StatusConflict
+
+	return nil, nil, http.StatusOK
+}
+
+func createInterfaces(id int, interfaces []tc.ServerInterfaceInfo, tx *sql.Tx) (error, error, int) {
+	ifaceQry := `
+	INSERT INTO interface (
+		max_bandwidth,
+		monitor,
+		mtu,
+		name,
+		server
+	) VALUES
+	`
+	ipQry := `
+	INSERT INTO ip_address (
+		address,
+		gateway,
+		interface,
+		server,
+		service_address
+	) VALUES
+	`
+
+	ifaceQueryParts := make([]string, 0, len(interfaces))
+	ipQueryParts := make([]string, 0, len(interfaces))
+	ifaceArgs := make([]interface{}, 0, len(interfaces))
+	ipArgs := make([]interface{}, 0, len(interfaces))
+	for i, iface := range interfaces {
+		argStart := i * 5
+		ifaceQueryParts = append(ifaceQueryParts, fmt.Sprintf("($%d, $%d, $%d, $%d, $%d)", argStart+1, argStart+2, argStart+3, argStart+4, argStart+5))
+		ifaceArgs = append(ifaceArgs, iface.MaxBandwidth, iface.Monitor, iface.MTU, iface.Name, id)
+		for _, ip := range iface.IPAddresses {
+			argStart = len(ipArgs)
+			ipQueryParts = append(ipQueryParts, fmt.Sprintf("($%d, $%d, $%d, $%d, $%d)", argStart+1, argStart+2, argStart+3, argStart+4, argStart+5))
+			ipArgs = append(ipArgs, ip.Address, ip.Gateway, iface.Name, id, ip.ServiceAddress)
 		}
 	}
 
-	current := TOServer{}
-	err := s.ReqInfo.Tx.QueryRowx(selectV20UpdatesQuery()+` WHERE sv.id=$1`, strconv.Itoa(*s.ID)).StructScan(&current)
+	ifaceQry += strings.Join(ifaceQueryParts, ",")
+	log.Debugf("Inserting interfaces for new server, query is: %s", ifaceQry)
+
+	ifaceRows, err := tx.Query(ifaceQry, ifaceArgs...)
```

Comment:
This seems like it should be an `Exec` instead if we really only care about the number of rows affected.

Reply:
TBH I don't even really care about the rows affected, I just ran into problems with not exhausting the rows so I counted them.